lib/engine/game/g_18_co.rb

```diff
@@ -167,6 +172,15 @@ def new_auction_round
         ])
       end
 
+      def action_processed(action)
+        super
+
+        case action
+        when Action::BuyCompany
+          mine_update_text(action.entity) if action.company == imc && action.entity.corporation?
```

Comment:
When the IMC is bought in, the text on the Ability needs to be updated to reflect the change in mine value and total.